Sources/DistributedCluster/Plugins/ClusterSystem+Plugins.swift

```diff
@@ -114,3 +114,9 @@ internal struct AnyPluginKey: Hashable, CustomStringConvertible {
         }
     }
 }
+
+/// Way to hook into ClusterSystem's actor lifecycle events, specifically to `ready` and `resign`.
+public protocol PluginActorLifecycleHook {
+    func actorReady<Act: DistributedActor>(_ actor: Act) where Act: DistributedActor, Act.ID == ClusterSystem.ActorID
+    func resignID(_ id: ClusterSystem.ActorID)
+}
```

Comment:
Explicitly made non as it's called synchronously in system.
This means though that if journal is an actor—those functions will be nonisolated. Thinking now if it's ok or not 🤔

Reply:
yeah those funcs being synchronous is good -- the plugin may have to use external synchronization, that's ok I guess -- we don't have a choice since the hooks are synchronous.

Naming nitpick, let's call them `onActorReady` and `onResignID`